### PR TITLE
Celtpipe fixes & pipes use planes to their full extend.

### PR DESCRIPTION
**Pull request by PJB3005**
Fixes a variety of bugs with celtpipes, can't be bothered to list them though.

Other things:

Pipes don't look derpy as crap anymore when layered and being partially hidden by floor tiles, vents now look ~fancy~ because their bezel is below the floors, even when the vent itself isn't.

New pipe layer manifold sprite, by @Skullyton, layer manifold now 100% supports celtpipes and looks sick:

Fixed unconnected pipe nodes not showing up on some things.

Made the layer adapter be coloured in the RPD, doesn't make its pipe item coloured though, nor fixes the horrible shading issues due to N3X15's sloth.

### CODE REVIEW

File: code/ATMOSPHERICS/pipes.dm [hunk 1]
```diff
@@ -604,12 +600,8 @@
 	color = PIPE_COLOR_GREEN
 /obj/machinery/atmospherics/pipe/manifold/insulated
 	name = "\improper Insulated pipe"
-	//icon = 'icons/obj/atmospherics/red_pipe.dmi'
-	icon_state = "manifold"
```
Comment (ghost) [inline code/ATMOSPHERICS/pipes.dm:608]: This is the only line in this PR that does what you say it does. The rest is not only non atomic but it'll cause bugs as well - for example, I'm pretty sure, and don't quote me on this, but you won't be able to colour insulated pipes using the RPD.

Comment (PJB3005) [reply code/ATMOSPHERICS/pipes.dm:608]: The rest of the PR was reducing var overrides that were already the same on the parent.

Comment (PJB3005) [reply code/ATMOSPHERICS/pipes.dm:608]: And yes I DID TEST colouring insulated pipes.

Comment (ghost) [reply code/ATMOSPHERICS/pipes.dm:608]: I'm too tired. This PR seems okay 👍

**Comment by PJB3005** [comment]
So I added a bunch of crap to this, can't be bothered to list it all because it fixes SO MUCH STUFF.

File: code/modules/RCD/schematics/pipe.dm [hunk 2]
```diff
@@ -485,7 +485,9 @@ var/global/list/disposalpipeID2State = list(
 
 /datum/rcd_schematic/pipe/layer_adapter/register_icon(var/dir)
 	for(var/layer = PIPING_LAYER_MIN to PIPING_LAYER_MAX)
-		register_asset("RPD_[pipe_id]_[dir]_[layer].png", new/icon('icons/obj/atmospherics/pipe_adapter.dmi', "adapter_[layer]", dir))
+		var/icon/I = new/icon('icons/obj/atmospherics/pipe_adapter.dmi', "adapter_[layer]", dir)
+		I.Blend(PIPE_COLOR_GREY)
```
Comment (PJB3005) [inline code/modules/RCD/schematics/pipe.dm]: This breaks it and makes the state in question completely white, needs to be fixed.

File: code/ATMOSPHERICS/pipes.dm [hunk 3]
```diff
@@ -999,17 +989,18 @@
 
 	..()
 
-/obj/machinery/atmospherics/pipe/layer_manifold/update_icon()
+/obj/machinery/atmospherics/pipe/layer_manifold/icon_directions()
+	return list(turn(dir, 180))
+
+/obj/machinery/atmospherics/pipe/layer_manifold/update_icon(var/adjacent_procd)
```
Comment (PJB3005) [inline code/ATMOSPHERICS/pipes.dm:1020]: I need to rewrite the entirety of this.

File: code/modules/RCD/schematics/pipe.dm (hunk 2 above)
Comment (ghost) [reply code/modules/RCD/schematics/pipe.dm]: > blend

There is no case where you need to use blend rather than color ever

Comment (PJB3005) [reply code/modules/RCD/schematics/pipe.dm]: Except there is because I need to send this icon COLOURED, this is for UI stuff.

Comment (ghost) [reply code/modules/RCD/schematics/pipe.dm]: Disgusting.

**Comment by Boxe** [comment]
revert hypothermia

**Comment by PJB3005** [comment]
Alright this should be done.

File: code/ATMOSPHERICS/atmospherics.dm [hunk 4]
```diff
@@ -68,68 +72,114 @@ Pipelines + Other Objects -> Pipe network
 		A.ex_act(severity)
 	..()
 
+/obj/machinery/atmospherics/proc/update_planes_and_layers()
+	return
+
+/obj/machinery/atmospherics/proc/icon_node_con(var/dir)
+	var/static/list/node_con = list(
```
Comment (clusterfack) [inline code/ATMOSPHERICS/atmospherics.dm]: Does this not create a new list every time?

**Comment by clusterfack** [comment]
Oh also conflicts somehow

File: code/ATMOSPHERICS/atmospherics.dm (hunk 4 above)
Comment (ComicIronic) [reply code/ATMOSPHERICS/atmospherics.dm:79]: I believe it does.

File: code/ATMOSPHERICS/atmospherics.dm [hunk 5]
```diff
@@ -68,68 +72,114 @@ Pipelines + Other Objects -> Pipe network
...
+	if (other.color)
+		return other.color
+
+	if (default_colour)
+		return default_colour
+
+	if (other.default_colour && other.default_colour != "#B4B4B4")
+		return other.default_colour
+
+	return "#B4B4B4"
```
Comment (ComicIronic) [inline code/ATMOSPHERICS/atmospherics.dm]: What's special about this colour?

File: code/ATMOSPHERICS/atmospherics.dm [hunk 6]
```diff
@@ -68,68 +72,114 @@ Pipelines + Other Objects -> Pipe network
...
+
+/obj/machinery/atmospherics/proc/node_layer()
+	return level == 1 ? PIPE_LAYER : EXPOSED_PIPE_LAYER
```
Comment (ComicIronic) [inline code/ATMOSPHERICS/atmospherics.dm]: What's special about level 1?

File: code/ATMOSPHERICS/atmospherics.dm (hunk 4 above)
Comment (PJB3005) [reply code/ATMOSPHERICS/atmospherics.dm:79]: It doesn't, I tested it once.

File: code/ATMOSPHERICS/atmospherics.dm (hunk 5 above)
Comment (PJB3005) [reply code/ATMOSPHERICS/atmospherics.dm]: Ask celt.

File: code/ATMOSPHERICS/atmospherics.dm (hunk 6 above)
Comment (PJB3005) [reply code/ATMOSPHERICS/atmospherics.dm]: Ask the oldcoders.

File: code/ATMOSPHERICS/atmospherics.dm (hunk 5 above)
Comment (ghost) [reply code/ATMOSPHERICS/atmospherics.dm]: That's what the old pipe colour was before the sprite was changed to white.

File: code/ATMOSPHERICS/atmospherics.dm (hunk 4 above)
Comment (clusterfack) [inline code/ATMOSPHERICS/atmospherics.dm]: How exactly did you test it

File: code/ATMOSPHERICS/atmospherics.dm (hunk 5 above)
Comment (ComicIronic) [reply code/ATMOSPHERICS/atmospherics.dm]: Do you know what I'm going to ask.

Comment (PJB3005) [reply code/ATMOSPHERICS/atmospherics.dm]: Yes

**Comment by clusterfack** [comment]
wee woo wee woo

**Comment by PJB3005** [comment]
I am aware

**Comment by PJB3005** [comment]
Compile failure and some bugs I missed fixed.

Ready for merge.

**Pull request merged**